expression/builtin_string_test.go

```diff
@@ -1203,3 +1203,31 @@ func (s *testEvaluatorSuite) TestElt(c *C) {
 		c.Assert(r, testutil.DatumEquals, types.NewDatum(t.ret))
 	}
 }
+
+func (s *testEvaluatorSuite) TestQuote(c *C) {
+	defer testleak.AfterTest(c)()
+
+	tbl := []struct {
+		arg interface{}
+		ret interface{}
+	}{
+		{`Don\'t!`, `'Don\\\'t!'`},
+		{`Don't`, `'Don\'t'`},
+		{`Don"`, `'Don"'`},
+		{`Don\"`, `'Don\\"'`},
+		{`\'`, `'\\\''`},
+		{`\"`, `'\\"'`},
+		{`萌萌哒(๑•ᴗ•๑)😊`, `'萌萌哒(๑•ᴗ•๑)😊'`},
+		{`㍿㌍㍑㌫`, `'㍿㌍㍑㌫'`},
```

Comment:
Add this to the case:
```
{string([]byte{0, 26}), `'\0\Z'`}
```

If you'd like, you could test those cases manually, compare with the result MySQL's behavior.
Pay attention to all escape char in the table https://dev.mysql.com/doc/refman/5.7/en/string-literals.html

For example:

```
select quote('\b'), quote('\Z'), quote('\r'), quote('\n'), quote('\t');
select quote('Don\'t'), quote('Don\\t'), quote('Don\\\'t');
```

rest LGTM